pipeline/metadata/flatten_satellite.py

```diff
@@ -371,54 +464,60 @@ def _process_satellite_v2_control(
         Rows
     """
     responses = scan.get('response', [])
-    if responses:
-      # An overall satellite v2 measurement
-      # always contains some non-control trial domains
-      is_control_domain = False
+
+    for response in responses:
+      is_control_domain = flatten_base.is_control_url(response['url'])
 
       row = {
           'domain':
-              scan['test_url'],
+              response['url'],
           'is_control':
               is_control_domain,
           'category':
-              self.category_matcher.get_category(scan['test_url'],
+              self.category_matcher.get_category(response['url'],
                                                  is_control_domain),
           'ip':
               scan['vp'],
           'is_control_ip':
               True,
           'date':
-              responses[0]['start_time'][:10],
+              response['start_time'][:10],
           'start_time':
-              format_timestamp(responses[0]['start_time']),
+              format_timestamp(response['start_time']),
           'end_time':
-              format_timestamp(responses[-1]['end_time']),
+              format_timestamp(response['end_time']),
           'anomaly':
               None,
           'success':
               not scan['connect_error'],
           'controls_failed':
               not scan['passed_control'],
-          'rcode': [str(response['rcode']) for response in responses],
+          'rcode': [str(response['rcode'])],
+          'has_type_a':
+              response['has_type_a'],
           'measurement_id':
               random_measurement_id,
           'source':
               flatten_base.source_from_filename(filepath),
       }
-      errors = [
-          response['error']
-          for response in responses
-          if response['error'] and response['error'] != 'null'
-      ]
-      row['error'] = ' | '.join(errors) if errors else None
-      for response in responses:
-        if response['url'] == row['domain']:
-          # Check response for test domain
-          if response['rcode'] == 0 and response['has_type_a']:
-            # Valid answers
-            row['has_type_a'] = True
-            # Separate into one answer IP per row for tagging
-            for ip in response['response']:
-              row['received'] = {'ip': ip}
-              yield row.copy()
+
+      if response['error'] == 'null':
+        error = None
+      else:
+        error = response['error']
+      row['error'] = error
+
+      received_ips = response['response']
```

Comment:
I don't see `response` in `ResponsesEntry.response`.  Did you mean `answer`?

Reply:
Same as above. Here are the example test with the `response.response` field

https://github.com/censoredplanet/censoredplanet-analysis/blob/534652d4ea0209d49a176e1c9ddbee5bfeae7a67/pipeline/metadata/test_flatten_satellite.py#L189